File: ios/RoomScanner/RoomScannerViewManager.m
//
//  RoomScannerViewManager.m
//  CozyAI
//
//  Created by God of Code on 5/14/25.
//

#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(RoomScannerViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onReady, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onMeshUpdate, RCTDirectEventBlock)

@end

File: ios/RoomScanner/RoomScannerViewManager.swift
//
//  RoomScannerViewManager.swift
//  CozyAI
//
//  Created by God of Code on 5/14/25.
//

import Foundation
import React

@objc(RoomScannerViewManager)
class RoomScannerViewManager: RCTViewManager {
  override func view() -> UIView! {
    return RoomScannerView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc func startSession(_ reactTag: NSNumber) {
    bridge.uiManager.addUIBlock { (uiManager, viewRegistry) in
      guard let view = viewRegistry?[reactTag] as? RoomScannerView else { return }
      view.startSession()
    }
  }

  @objc func stopSession(_ reactTag: NSNumber) {
    bridge.uiManager.addUIBlock { (uiManager, viewRegistry) in
      guard let view = viewRegistry?[reactTag] as? RoomScannerView else { return }
      view.stopSession()
    }
  }
}
